aten/src/ATen/native/Loss.cpp

```diff
@@ -79,7 +79,10 @@ Tensor margin_ranking_loss(const Tensor& input1, const Tensor& input2, const Ten
 Tensor kl_div(const Tensor& input, const Tensor& target, int64_t reduction) {
   auto output_pos = target * (at::log(target) - input);
   auto zeros = at::zeros_like(output_pos, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+  // Perform a check for (target > -EPSILON) & (target < EPSILON) since floating point
+  // errors in output_pos cause the output values to be 1e-17 on GPU and -1e-14 on CPU.
   auto output = at::where(target > 0, output_pos, zeros);
+  output = at::where(output < EPSILON, zeros, output);
```

Comment:
This is a little ad-hoc and, as your PR description comment indicates, likely better dealt with the same way as gh-31378 (`cosine_similarity`). I suggest to wait till the `cosine_similarity` PR is merged, and then copy that same strategy. Don't try to copy it now, because that PR is still in flux and it'd result in double work.

Reply:
@v0dro you could also have a chat with @nikitaved about this. I suspect the `cosine_similarity` fix is the right way to go, so if you two can coordinate to push this forward that'd be useful.

Reply:
Alright. I think I'll wait until that PR is merged, understand the methodology and then use that here.